### PR TITLE
add ts optional property

### PR DESCRIPTION
**Pull request by ZeevGerstner**
### Related issues
#2792

### Short description and why it's useful

Added ts optional attributes (?) to catalog/product.ts, that not includes in

https://github.com/DivanteLtd/vue-storefront-integration-boilerplate/blob/master/1.%20Expose%20the%20API%20endpoints%20required%20by%20VS/Required%20API%20specification.md#get-vsbridgeproductsindex

in products/index

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [ ] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [ ] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [ ] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

File: core/modules/catalog/types/Product.ts [hunk 1]
```diff
@@ -1,62 +1,62 @@
...
+  color_options?: Array<number>
   configurable_children: Array<Object>
   configurable_options: Array<Object>
-  custom_attributes: any
+  custom_attributes?: any
   description: string
-  eco_collection: string
-  erin_recommends: string
-  errors: Object
+  eco_collection?: string
```
Comment (pkarw) [inline core/modules/catalog/types/Product.ts]: I belive this two should be removed from the schema they’re just specific to magento demo data

File: core/modules/catalog/types/Product.ts [hunk 2]
```diff
@@ -1,62 +1,62 @@
 export default interface Product {
   category: Array<Object>
   category_ids: Array<string>
-  climate: string
+  climate?: string
```
Comment (pkarw) [inline core/modules/catalog/types/Product.ts]: Should be removed - specific to just the demo data set

File: core/modules/catalog/types/Product.ts [hunk 3]
```diff
@@ -1,62 +1,62 @@
...
+  msrp_display_actual_price_type?: string
   name: string
   new: string
-  options: Array<Object>
-  parentSku: string
-  pattern: string
-  performance_fabric: string
+  options?: Array<Object>
+  parentSku?: string
+  pattern?: string
```
Comment (pkarw) [inline core/modules/catalog/types/Product.ts]: Performace_fabric should be removed at all

File: core/modules/catalog/types/Product.ts (hunk 1 above)
Comment (ZeevGerstner) [reply core/modules/catalog/types/Product.ts]: So you want me to remove eco_collection & erin_recommends from the file?

File: core/modules/catalog/types/Product.ts (hunk 3 above)
Comment (pkarw) [reply core/modules/catalog/types/Product.ts]: Correct 👍

Comment (pkarw) [reply core/modules/catalog/types/Product.ts]: Please do also update the changelog

**Comment by ZeevGerstner** [comment]
Removed the unnecessary attributes. Hope its ok now :)

**Comment by pkarw** [comment]
Thanks!

**Comment by ZeevGerstner** [comment]
Cheers mate, glad to help. Can you please add me to the contributors list?

**Pull request merged**